### PR TITLE
sys-kernel/coreos-sources: Accomodate the patch to apply on latest kernel

### PR DESCRIPTION
**Pull request by sayanchowdhury**
# sys-kernel/coreos-sources: Accomodate the patch to apply on latest kernel

As the repo was huge, I clone it in the devbox, and seems like the editor was not setup well hence the format issues in patch but emerge works well.

## Testing done

[Describe the testing you have done before submitting this PR. Please include both the commands you issued as well as the output you got.]

- [ ] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)
- [ ] Inspected CI output for image differences: `/boot` and `/usr` size, packages, list files for any missing binaries, kernel modules, config files, kernel modules, etc.

### CODE REVIEW

**Comment by sayanchowdhury** [comment]
CI http://jenkins.infra.kinvolk.io:8080/job/container/job/packages_all_arches/3425/cldsv/

File: sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.1/z0002-Revert-scripts-pahole-flags.sh-Parse-DWARF-and-gener.patch [hunk 1]
```diff
@@ -1,32 +1,30 @@
...
  if [ "${pahole_ver}" -ge "121" ]; then
- 	extra_paholeopt="${extra_paholeopt} --btf_gen_floats"
+        extra_paholeopt="${extra_paholeopt} --btf_gen_floats"
  fi
 -if [ "${pahole_ver}" -ge "122" ]; then
--	extra_paholeopt="${extra_paholeopt} -j"
+-       extra_paholeopt="${extra_paholeopt} -j"
 -fi
- 
- echo ${extra_paholeopt}
```
Comment (pothos) [inline sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.1/z0002-Revert-scripts-pahole-flags.sh-Parse-DWARF-and-gener.patch:29]: Why is this removed?

File: sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.1/z0002-Revert-scripts-pahole-flags.sh-Parse-DWARF-and-gener.patch [hunk 2]
```diff
@@ -1,32 +1,30 @@
...
--- 
+ if [ "${pahole_ver}" -ge "124" ]; then
+        # see PAHOLE_HAS_LANG_EXCLUDE
+        extra_paholeopt="${extra_paholeopt} --lang_exclude=rust"
+--
 2.25.1
```
Comment (pothos) [inline sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.1/z0002-Revert-scripts-pahole-flags.sh-Parse-DWARF-and-gener.patch:30]: The new patch we have also covers pahole version 125, does it make sense to use that?
https://github.com/flatcar/scripts/blob/main/sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.6/z0002-revert-pahole-flags.patch

File: sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.1/z0002-Revert-scripts-pahole-flags.sh-Parse-DWARF-and-gener.patch (hunk 1 above)
Comment (pothos) [reply sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.1/z0002-Revert-scripts-pahole-flags.sh-Parse-DWARF-and-gener.patch:29]: Removing this looks like it will make the patch have no effect

File: sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.1/z0002-Revert-scripts-pahole-flags.sh-Parse-DWARF-and-gener.patch (hunk 2 above)
Comment (pothos) [reply sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.1/z0002-Revert-scripts-pahole-flags.sh-Parse-DWARF-and-gener.patch:30]: Probably yes, because I see 1.25 in this list: https://beta.release.flatcar-linux.net/amd64-usr/current/flatcar_developer_container_packages.txt

**Comment by ader1990** [comment]
Had to do the same thing for the 6.6, maybe this part here helps: https://github.com/ader1990/scripts/blob/f1c8d36616ec4902924644e9cc89f9aba6bce736/sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.6/z0002-revert-pahole-flags.patch

File: sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.1/z0002-Revert-scripts-pahole-flags.sh-Parse-DWARF-and-gener.patch (hunk 1 above)
Comment (jepio) [reply sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.1/z0002-Revert-scripts-pahole-flags.sh-Parse-DWARF-and-gener.patch:29]: this is the context line. it is lower down in the file now, the new context is green 26-29.

Comment (jepio) [reply sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.1/z0002-Revert-scripts-pahole-flags.sh-Parse-DWARF-and-gener.patch:29]: reading diffs of patches is tricky 😄

Comment (pothos) [reply sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.1/z0002-Revert-scripts-pahole-flags.sh-Parse-DWARF-and-gener.patch:29]: Ah, that other minus :D
Anyway, I think we can try to use the same patch we have in main, or?

**Review by pothos — COMMENTED** [review]
Would be good to link to the file in main or state that it is the same

File: sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/coreos-sources-6.1.77.ebuild [hunk 3]
```diff
@@ -36,5 +36,5 @@ IUSE=""
 # local patches overlap with the upstream patch.
 UNIPATCH_LIST="
```
Comment (pothos) [inline sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/coreos-sources-6.1.77.ebuild:37]:
```suggestion
# revert-pahole-flags is the same as https://github.com/flatcar/scripts/blob/1c5e1909304eef5d1c96a80b944db02701294d24/sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.6/z0002-revert-pahole-flags.patch
UNIPATCH_LIST="
```

**Review by pothos — APPROVED** [review]
assuming CI passes
